torch/_inductor/codegen/triton.py

```diff
@@ -296,14 +296,46 @@ def log10(x):
     def nextafter(x, y):
         return f"tl.math.nextafter({x}, {y})"
 
+    @staticmethod
+    def logical_not(a):
+        return f"{a} ^ 1"  # !{a} is not recognized by Triton
+
     @staticmethod
     def logical_and(a, b):
-        return f"{a} & {b}"
+        return f"{a} & {b}"  # {a} && {b} is not recognized by Triton
 
     @staticmethod
     def logical_or(a, b):
+        return f"{a} | {b}"  # {a} || {b} is not recognized by Triton
+
+    @staticmethod
+    def logical_xor(a, b):
+        return f"{a} ^ {b}"
```

Comment:
Also, due to precedence, you may need to put parens on these? I believe that there's a function to conditionally put parens on an expression if it doesn't have it yet.